tests/test.yml

```diff
@@ -22,51 +22,14 @@
-       - {name: admin7, state: 'absent', uid: 5007, group: "{{admingroup}}", shell: "{{adminshell}}", pubkey: "ssh-rsa KEY admin7@example.com", groups: "agroup,bgroup", comment: "commentB" }
-       - {name: admin8, state: 'present', uid: 5008, group: "{{admingroup}}", shell: "{{adminshell}}", pubkey: "ssh-rsa KEY admin8@example.com", groups: "bgroup", comment: "commentB", remove: "no" }
-       - {name: admin9, state: 'absent', uid: 5009, group: "{{admingroup}}", shell: "{{adminshell}}", groups: "bgroup", comment: "commentC", remove: "yes" }
-       - {name: admin10, state: 'present', uid: 5010, group: "{{admingroup}}", shell: "{{adminshell}}", pubkey: "ssh-rsa KEY admin10@example.com", key_state: "absent" }
-       - {name: admin11, state: 'present', uid: 5011, group: "{{admingroup}}", shell: "{{adminshell}}", home: "/var/lib/admin11", create_home: "yes" }
-       - {name: admin12, state: 'present', uid: 5012, password: "$6$1lg6WhuGmrG1QE5X$O9xMkmnsULiqDeZ3DdSKQrHcGeCW4U7FKLhWQIRQGKsG5sslmWUcgQLHz.T/lNqJhVmk8m9OJ2Iv.ogJn7x/" }
-     - moreusers:
-       - {name: user1, state: 'present', uid: 6001, shell: "{{adminshell}}" }
-       - {name: user2, state: 'present', uid: 6002 }
-       - {name: user3, state: 'absent', uid: 6003 }
-       - {name: user4, state: 'absent', uid: 6004, pubkey: "ssh-rsa KEY user4@example.com", remove: "yes" }
-       - {name: user5, state: 'present', uid: 6005, pubkey: "ssh-rsa KEY5 user5@example.com" }
-       - {name: user6, state: 'present', uid: 6006, pubkey: "ssh-rsa KEY6 user6@example.com", options: 'command="/usr/local/bin/rrsync /allow/rrsync/here/directory",no-agent-forwarding,no-port-forwarding,no-pty,no-user-rc,no-X11-forwarding' }
-       - {name: user7, state: 'absent', uid: 6007, pubkey: "ssh-rsa KEY6 user6@example.com", options: 'command="/usr/local/bin/rrsync /allow/rrsync/here/directory",no-agent-forwarding,no-port-forwarding,no-pty,no-user-rc,no-X11-forwarding' }
-       - {name: user8, state: 'absent', uid: 6008, comment: "comment1", remove: "yes" }
-       - {name: user9, state: 'present', uid: 6009 }
-       - {name: user10, state: 'present', uid: 6010, group: "{{admingroup}}", groups: "agroup" }
-     - anotheruserlisthere:
-       - {name: training1, state: 'present', uid: 7001, shell: "{{adminshell}}" }
-       - {name: training2, state: 'present', uid: 7002 }
-       - {name: training3, state: 'absent', uid: 7003 }
-     - admin_list_of_user_lists:
-       - "{{ adminusers | default([]) }}"
-       - "{{ moreusers | default([]) }}"
-       - "{{ anotheruserlisthere | default([]) }}"
- 
-     - admin_root_keys:
-       - { state: 'present', pubkey: "ssh-rsa KEY admin1@example.com" }
-       - { state: 'absent', pubkey: "ssh-rsa KEY2 bad_admin2@example.com" }
-     - admin_list_of_groups:
-       - { name: agroup, gid: 27001 }
-       - { name: bgroup, gid: 27002 }
-     
-   roles:
-     - ansible-role-users
+   post_tasks:
```

Comment:
I'll take this learning opportunity. I don't understand why this post_task is needed. What would happen if it was removed? Could it have another name or comment?

Reply:
If it's removed then ansible I think will complain about syntax errors because the task to kill the screen is not in a list of tasks.
It could be changed to tasks (or pre_tasks).
I don't know which comment you refer to so can't comment on that ;))

Reply:
What I meant to ask was: Why do we need to kill all the screen sessions?

Reply:
Ah. If we don't then there's a 'user logged in', and the role runs the usermod way of changing said user ; which always(as per the task config) results in a run that has changed something.

Reply:
Which brings the next good question: why do those tasks change needlessly? :)

One possible improvement to this role would be to add more conditionals to only change/run the task [1] if the GID/UID of the user (and of the logged in process or is that overkill?) are different than what is in code. If the usermod command has some nice return codes one could use those but after some searching, I couldn't find any good ones  [2]

[1]: https://github.com/CSCfi/ansible-role-users/blob/master/tasks/main.yml#L115-L137
[2]: https://github.com/shadow-maint/shadow/blob/master/src/usermod.c#L81

Reply:
Okay, I think I remember hearing about this before. I think I now understand everything ™... ...merging.

Reply:
👌 I created #51